recipe/meta.yaml

```
@@ -10,32 +11,36 @@ package:
   version: {{ version }}
 
 source:
-  # we have to use git urls here to get the submodules needed for the
-  # build
-  git_url: https://github.com/dmlc/xgboost
-  git_tag: v{{ version }}
+  url: https://github.com/dmlc/xgboost/releases/download/v{{ version }}/xgboost-{{ version }}.tar.gz
+  sha256: 28bec8e821b1fefcea722d96add66024adba399063f723bc5c815f7af4a5f5e4
   patches:
     # xgboost patches
-    - 0001-Force-endian-flag-in-cross-compilation-mode.patch  # [arm64 or aarch64 or ppc64le]
-    - 0002-Enable-latest-libcxx-on-MacOS.patch  # [osx]
-    - 0003-Use-mingw-w64-path.patch
-    - 0004-Undo-dmlc-xgboost-9436.patch
+    - patches/0001-Force-endian-flag-in-cross-compilation-mode.patch  # [arm64 or aarch64 or ppc64le]
+    - patches/0002-Enable-latest-libcxx-on-MacOS.patch  # [osx]
+    - patches/0003-Use-mingw-w64-path.patch
+    - patches/0004-Undo-dmlc-xgboost-9436.patch
+    - patches/0005-CMake-Explicitly-link-with-CCCL-standalone-or-CTK.patch
 
 build:
   number: {{ build_number }}
-  skip: True  # [cuda_compiler_version == "11.2"]
+  {% if python is defined %}
+  skip: {{ python.split(".")[:2] != min_python.split(".")[:2] }}
+  {% endif %}
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}  # [cuda_compiler != "None" and cuda_compiler_version != "None"]
+    - {{ compiler('m2w64_c') }}    # [win]
+    - {{ compiler('m2w64_cxx') }}  # [win]
+    - {{ compiler('cuda') }}       # [cuda_compiler != "None" and cuda_compiler_version != "None"]
     - {{ stdlib('c') }}
     - cmake
     - llvm-openmp  # [osx]
     - libgomp      # [linux]
     - ninja
   host:
+    - cccl         # [cuda_compiler != "None"]
     - nccl         # [linux and cuda_compiler != "None"]
```

This will exclude both of the problematic `cccl` versions as discussed above. If we need to exclude more, we can modify this further

```suggestion
    - cccl !=2.4.0,!=2.5.0  # [cuda_compiler != "None"]
    - nccl                  # [linux and cuda_compiler != "None"]
```